lib/internal/modules/esm/loader.js

```diff
@@ -116,21 +156,25 @@ class ESMLoader {
   translators = translators;
 
   constructor() {
-    if (getOptionValue('--experimental-loader')) {
+    if (getOptionValue('--experimental-loader').length > 0) {
       emitExperimentalWarning('Custom ESM Loaders');
     }
     if (getOptionValue('--experimental-network-imports')) {
       emitExperimentalWarning('Network Imports');
     }
-    if (getOptionValue('--experimental-specifier-resolution') === 'node' && !emittedSpecifierResolutionWarning) {
+    if (getOptionValue('--experimental-specifier-resolution') === 'node') {
       process.emitWarning(
         'The Node.js specifier resolution flag is experimental. It could change or be removed at any time.',
         'ExperimentalWarning'
       );
-      emittedSpecifierResolutionWarning = true;
```

Comment:
Can we put back the `emittedSpecifierResolutionWarning` stuff? I assume you took it out earlier as you were trying to debug the error?

Reply:
There should be a test failing if it was indeed important. Maybe we don't need it after all, the constructor should be called only once, right?

Reply:
That’s a good question.

Reply:
There was a test for it (test/es-module/test-esm-experimental-warnings.mjs). It didn't fail because the subsequent commit adding the "nothing" case was lost; I'm adding it back now (I kept a backup of the branch pre-flattening).

Until we have a better idea, I'd like to put back the `isInternal` (which Targos/Michaël removed with his fix for the async_hook issue). My rationale is:

* these warnings are for end-user consumption and are not relevant for whatever Node.js is doing internally (so they shouldn't be emitted until something happens in user-land)
* it _dramatically_ reduces the amount of noise in test output
* as long as the flag is used only to toggle non-critical behaviour (like emitting warnings) and NOT to change fundamental behaviour (do load this vs don't load this), it should be safe. It does require a bit of human policing, but I think we all would agree to block using that flag to change fundamental behaviour unless there was a very, very good reason.
* (AFAIK) there is no other way to affect only user-land
* (least importantly) it's quite handy when troubleshooting: `if (!isInternal) console.log(…)` you almost never want to log for the internal instance of ESMLoader

Reply:
I removed `isInternal` because of that comment (and it introduced no test failures):
![image](https://user-images.githubusercontent.com/2352663/166153165-343b361b-66b0-4aa8-bac8-f97504b2d448.png)

Reply:
Since we don't need it for this PR, can we keep it removed and then you can propose adding it in its own PR?

Reply:
We need _something_, and `isInternal` addresses the issue more completely than the `let emitted…` in that nothing is printed until user-land is reached.

I'm not at all married to `isInternal` btw; it just seems the best option so far.

Reply:
The flag variable we had is the status quo, so let's keep that and change it in its own PR. This PR is huge, so I think we should exclude any changes that aren't required to support chaining.